crates/bevy_reflect/src/impls/std.rs

```diff
@@ -591,6 +594,19 @@ mod tests {
         let _serializable = reflect_serialize.get_serializable(&std::time::Duration::ZERO);
     }
 
+    #[test]
+    fn can_serialize_phantomdata() {
+        let mut type_registry = TypeRegistry::default();
+        type_registry.register::<std::marker::PhantomData<()>>();
+
+        let reflect_serialize = type_registry
+            .get_type_data::<ReflectSerialize>(
+                std::any::TypeId::of::<std::marker::PhantomData<()>>(),
+            )
+            .unwrap();
+        let _serializable = reflect_serialize.get_serializable(&std::marker::PhantomData::<()>);
```

Comment:
I think it would be better to serialize to a string and compare the output with an `assert_eq!`. That way, we can actually test that the serialized version looks like it should (I suspect this is one of the places where the `ReflectSerializer` might fail).

Also can we put the `PhantomData` in a struct? Since that's its most likely use case.